src/components/SearchableSingleSelect/SearchableSingleSelect.tsx

```diff
@@ -0,0 +1,175 @@
+import i18n from '@dhis2/d2-i18n'
+import {
+    CircularLoader,
+    Input,
+    SingleSelect,
+    SingleSelectOption,
+} from '@dhis2/ui'
+import React, { forwardRef, useCallback, useEffect, useState } from 'react'
+import { useDebouncedCallback } from 'use-debounce'
+import classes from './SearchableSingleSelect.module.css'
+
+interface Option {
+    value: string
+    label: string
+}
+
+const Loader = forwardRef<HTMLDivElement, object>(function Loader(_, ref) {
+    return (
+        <div ref={ref} className={classes.loader}>
+            <CircularLoader />
+        </div>
+    )
+})
+
+function Error({
+    msg,
+    onRetryClick,
+}: {
+    msg: string
+    onRetryClick: () => void
+}) {
+    return (
+        <div className={classes.error}>
+            <div className={classes.errorInnerWrapper}>
+                <span className={classes.loadingErrorLabel}>{msg}</span>
+                <button
+                    className={classes.errorRetryButton}
+                    type="button"
+                    onClick={onRetryClick}
+                >
+                    {i18n.t('Retry')}
+                </button>
+            </div>
+        </div>
+    )
+}
+
+type OnChange = ({ selected }: { selected: string }) => void
+type OnFilterChange = ({ value }: { value: string }) => void
+type OnIntersectionChange = ({
+    isIntersecting,
+}: {
+    isIntersecting: boolean
+}) => void
+interface SearchableSingleSelectPropTypes {
+    onChange: OnChange
+    onFilterChange: OnFilterChange
+    onIntersectionChange: OnIntersectionChange
+    onRetryClick: () => void
+    options: Option[]
+    preventIntersectionDetection: boolean
+    showEndLoader: boolean
+    loading: boolean
+    selected?: string
+    error?: string
+}
+
+export const SearchableSingleSelect = ({
+    error,
+    loading,
+    onChange,
+    onFilterChange,
+    onIntersectionChange,
```

Comment:
Would it make sense to align this with transfer-prop? `onEndReached`? We're not actually interested in the "change", right? We just want to call the callback when it's reached, eg. `isIntersecting: true`